Library/Homebrew/cmd/install.rb

```diff
@@ -196,6 +196,19 @@ def install
       )
     end
 
+    # Cover confusion from editing formulae on an install updated to 4.0.0.
+    if args.build_from_source? && !Homebrew::EnvConfig.no_install_from_api? && !Homebrew::EnvConfig.developer? &&
+       !Homebrew::EnvConfig.no_env_hints? && formulae.any?(&:core_formula?) && CoreTap.instance.installed?
```

Comment:
Feels like this should maybe be in `Library/Homebrew/install.rb` so it can be used in both `cmd/upgrade.rb` and `cmd/reinstall.rb`.